subsys/bootloader/bl_crypto/bl_crypto_client.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bl_crypto.h>
+#include "bl_crypto_internal.h"
+#include <fw_metadata.h>
+#include <kernel.h>
+
+extern struct fw_abi_getter_info abi_getter_in;
+void print_header(const struct bl_crypto_abi * obj){
+	printk("====HEADER====\n\r");
+	for (u32_t i = 0; i < 3; i++) {
+		printk("Magic values: 0x%x\n\r", obj->header.magic[i]);
+	}
+	printk("Flags: %d and version: %d\n\r", obj->header.abi_flags, 	obj->header.abi_version);
+	printk("ABI_id: 0x%x\n\r", obj->header.abi_id);
+	printk("ABI len: %d\n\r", obj->header.abi_len);
+}
+
+const struct bl_crypto_abi* get_bl_crypto_abi(void)
+{
+	//printk("Getting crypto ABI.\n");
+	static const struct fw_abi_info * bl_crypto_abi = NULL;
+	if (!bl_crypto_abi) {
+		int ret = abi_getter_in.abi_getter(0x12345678, 0, &bl_crypto_abi);
+		if (ret) k_oops();
+	}
+	//print_header((const struct bl_crypto_abi *)bl_crypto_abi);
+	return (const struct bl_crypto_abi *)bl_crypto_abi;
+}
+
+int crypto_root_of_trust(const u8_t *pk, const u8_t *pk_hash,
+			 const u8_t *sig, const u8_t *fw,
+			 const u32_t fw_len)
+{
+	return get_bl_crypto_abi()->abi.crypto_root_of_trust(pk, pk_hash, sig, fw, fw_len);
+}
+
+bool verify_sig(const u8_t *data, u32_t data_len, const u8_t *sig,
+		const u8_t *pk)
+{
+	return get_bl_crypto_abi()->abi.verify_sig(data, data_len, sig, pk);
+}
+
+int bl_sha256_init(bl_sha256_ctx_t * ctx)
+{
+	if (sizeof(*ctx) < get_bl_crypto_abi()->abi.bl_sha256_ctx_size) {
+		return -EFAULT;
+	}
+	return get_bl_crypto_abi()->abi.bl_sha256_init(ctx);
+}
+
+int bl_sha256_update(bl_sha256_ctx_t * ctx, const u8_t * data, u32_t data_len)
+{
+	return get_bl_crypto_abi()->abi.bl_sha256_update(ctx, data, data_len);
+}
+
+int bl_sha256_finish(bl_sha256_ctx_t * ctx, u8_t * output)
+{
+	return get_bl_crypto_abi()->abi.bl_sha256_finish(ctx, output);
+}
+
+int bl_ecdsa_verify_secp256r1(const u8_t * hash, u32_t hash_len, const u8_t * public_key, const u8_t * signature)
```

Comment:
This is mostly my fault for not following the file's convention. Anyways I think `pk` is a bad abbreviation as it could be `public` or `private` key so I would suggest having `pubk` and `privk`  or write out key `pubkey` if we are going to abbreviate.

Reply:
Done